packages/wire/lib/messages/OpenChannelMessage.ts

```diff
@@ -27,12 +27,12 @@ export class OpenChannelMessage implements IWireMessage {
         reader.readUInt16BE(); // read type
         instance.chainHash = reader.readBytes(32);
         instance.temporaryChannelId = reader.readBytes(32);
-        instance.fundingSatoshis = reader.readUInt64BE();
-        instance.pushMsat = reader.readUInt64BE();
-        instance.dustLimitSatoshis = reader.readUInt64BE();
-        instance.maxHtlcValueInFlightMsat = reader.readUInt64BE();
-        instance.channelReserveSatoshis = reader.readUInt64BE();
-        instance.htlcMinimumMsat = reader.readUInt64BE();
+        instance.fundingSatoshis = Value.fromSats(reader.readUInt64BE());
+        instance.pushMsat = Value.fromMilliSats(reader.readUInt64BE());
+        instance.dustLimitSatoshis = Value.fromSats(reader.readUInt64BE());
+        instance.maxHtlcValueInFlightMsat = Value.fromMilliSats(reader.readUInt64BE());
+        instance.channelReserveSatoshis = Value.fromSats(reader.readUInt64BE());
+        instance.htlcMinimumMsat = Value.fromMilliSats(reader.readUInt64BE());
         instance.feeRatePerKw = reader.readUInt32BE();
```

Comment:
`feeRatePerKw` should probably be a `Value` as well

Reply:
Maybe I am missing something but is it fine if I convert `bigint` to `numbers` in alias `Value` ? because if we look @ the js number range it can hold upto 2^53 which is approx. 10^15 roughly and in Satoshis if we consider the milli/sat/millionthsats/ its gonna be fine because all of them are under 10^15 but in case of picosats it > 10^15 around 10^20 (BigInt is required).

1 bitcoin = 10^8 satoshis
1 bitcoin = 10^11 millisatoshis
1 bitcoin = 10^14 millionthsatoshis

Reply:
Aah figured💡  evry time we take fee value we are converting it to `pico_sats` * 10^12 so if any fee inpt value greater than 10^3 it will lead to overflow. So `bigint` will get the job done can't use `number`.

Reply:
`Value` is really more about creating a complex type for Bitcoin value. This allows us to have a consistent way to store "money" without having to think about overflows or conversions between various fractional values.  Internally it stores it in picosats since that's the smallest value we'll ever use for Lightning.

Down the line we can also add math operations to the type so we do things like adding two values like sats and millisats without issue.  

More info: https://www.martinfowler.com/eaaCatalog/money.html